pkg/utils/utils.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import (
+	"crypto/sha512"
+	"fmt"
+)
+
+// 29 - len('CNI') - 2*len('-')
+const maxNameLen = 16
+
+// Generates a chain name to be used with iptables.
+// Ensures that the generated name is less than
+// 29 chars in length
+func FormatChainName(name string, id string) string {
+	h := sha512.Sum512([]byte(id))
+	if len(name) > maxNameLen {
+		return fmt.Sprintf("CNI-%s-%x", name[:len(name)-maxNameLen], h[:8])
```

Comment:
@achanda @alban PTAL at https://github.com/appc/cni/pull/168/files#diff-fb81b0a96eed5b010cfb081b3aef4431R15. It's a step closer but theoretically definitely not unique.